okhttp/src/jvmTest/java/okhttp3/WebPlatformToAsciiTest.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2023 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.TestFactory
+
+/** Runs the web platform ToAscii tests. */
+class WebPlatformToAsciiTest {
+  val knownFailures = setOf(
+    // OkHttp rejects empty labels.
+    "x..xn--zca",
+    "x..ß",
+
+    // OkHttp rejects labels longer than 63 code points, the web platform tests don't.
+    "x01234567890123456789012345678901234567890123456789012345678901x.xn--zca",
+    "x01234567890123456789012345678901234567890123456789012345678901x.ß",
+    "x01234567890123456789012345678901234567890123456789012345678901x",
+    "x01234567890123456789012345678901234567890123456789012345678901†",
+
+    // OkHttp rejects domain names longer than 253 code points, the web platform tests don't.
```

Comment:
Who is right?

Reply:
I think the issue is that the web platform tests cover a narrower feature set than HttpUrl. (Calling `new URL` in JavaScript is more lenient than OkHttp here.)